optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -332,11 +338,24 @@ public void setAssertionScoreDirectorFactory(ScoreDirectorFactoryConfig assertio
         this.assertionScoreDirectorFactory = assertionScoreDirectorFactory;
     }
 
+    /**
+     * @deprecated for removal.
+     */
+    @Deprecated(/* forRemoval = true */)
     public Boolean isGenerateDroolsTestOnError() {
         return generateDroolsTestOnError;
     }
 
+    /**
+     * @deprecated for removal.
+     */
+    @Deprecated(/* forRemoval = true */)
     public void setGenerateDroolsTestOnError(Boolean generateDroolsTestOnError) {
+        if (BooleanUtils.isTrue(generateDroolsTestOnError)) {
+            System.setProperty(GENERATE_DROOLS_TEST_ON_ERROR_PROPERTY_NAME, "true");
+        } else {
+            System.clearProperty(GENERATE_DROOLS_TEST_ON_ERROR_PROPERTY_NAME);
+        }
         this.generateDroolsTestOnError = generateDroolsTestOnError;
```

Comment:
This should be a simple setter (that is now deprecated). It shouldn't mess with the system property, in favor of which it is deprecated, because  System.setProperty() is dangerous: presume that a single JVM process runs 2 solver configs, constructed by java (not by xml) and the first one calls setGenerateDroolsTestOnError(true), but the second one doesn't. The second one would feel it too.